Code refactoring for application properties

pankaj-cgi opened the pull request:
Property injection was used in Route class to pass on the properties to other classes. Passing the values was divided in 2  types:
1. AuthorizationProperties was used as object of properties in application.properties
2. Some of the properties were passed directly to the constructors of required classes and some were set in AuthorizationProperties  and object of AuthorizationProperties  was passed.

In code refactoring, 
1. ApplicationProperty.java (enum) mapped to all the properties in application.properties. 
2. The properties from application.properties are loaded in ApplicationProperties object at startup of application. This object is available to all classes to read the property values from application.properties file. 
Cons:
* Single place holder for properties
* Injection happening at one place
* Properties are available to all classes, irrespective if it has access to CamelContext or not

daveb-hni commented:
- Routes.java
	Remove commented out code, complete TODO or update comment a little to clarify what needs to be done.

- ValidateAccessToken.java
	ln 65 
		compiler complains about generics on DefaultJOSEObjectTypeVerifier, may be able to use diamond operator to infer generics
		could add static String for hardcode JWT value.
	ln 91
		generics complaint, again diamond operator may work		

- ApplicationProperty.java	
	Do we want to name enums as *Enum.java for ease of identifying them? It's not Java convention but I found it useful in projects.
	
- V2PayloadValidator.java
	Unnecessary blank lines
	
- PopulateReqHeader
	Not from this commit but could we rename this as it is not Java naming convention of a noun (it's a method name) e.g. RequestHeaderPopulater

- Util.java	
	Again not from this commit but there are unused imports (can turn on Orgaize Imports on Save if using Eclipse or use Ctrl + Shift + O)
	Is it best to load from Util? 

- ValidateAccessTokenTest
	Unused properties

- V2PayloadValidatorTest

	Do we want a parent test class that calls cleanup and load for all classes?

- RouteTest.java
	Looks good!

pankaj-cgi commented:
Addressed comments:
Routes.java
Removed commented out code, removed completed TODO

ValidateAccessToken.java
- added static string for JWT
- Resolved warnings

ApplicationProperty.java
Do we want to name enums as *Enum.java for ease of identifying them? It's not Java convention but I found it useful in projects.
- I agree, we can post this question in Developers channel. If all are good we can rename Enums in client and hnsecure for consistency

V2PayloadValidator.java
- Removed blank lines

PopulateReqHeader
Not from this commit but could we rename this as it is not Java naming convention of a noun (it's a method name) e.g. 
RequestHeaderPopulater
- Did not resolve this one. We can resolve it with Enum renaming.

Util.java
- Organized imports
- Need to discuss: Is it best to load from Util?

ValidateAccessTokenTest
- Removed Unused properties

V2PayloadValidatorTest
Do we want a parent test class that calls cleanup and load for all classes?
- I agree. Currently only one test class is using so i did not add initially. But added now for future purposes.


RouteTest.java
Looks good!
